middleware/auth.go

```diff
@@ -41,11 +40,12 @@ func BasicAuth(fn BasicValidateFunc) echo.HandlerFunc {
 						if fn(cred[:i], cred[i+1:]) {
 							return nil
 						}
-						he.SetCode(http.StatusUnauthorized)
 					}
 				}
 			}
 		}
+		
+		c.Response().Header().Add("WWW-Authenticate", "Basic")
```

Comment:
We would like to header to be coming from constant, look at echo.go. "Basic" could also be a constant in the same context. I see people also using realm, check on that too.